cumulusci/tasks/salesforce/composite.py

```diff
@@ -0,0 +1,143 @@
+import json
+import random
+import re
+import string
+from pathlib import Path
+
+from cumulusci.cli.ui import CliTable
+from cumulusci.core.exceptions import SalesforceException
+from cumulusci.core.utils import process_bool_arg, process_list_arg
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+from cumulusci.utils import inject_namespace
+
+API_ROLLBACK_MESSAGE = "The transaction was rolled back since another operation in the same transaction failed."
+API_INVALID_REF_MESSAGE = "Invalid reference specified."
+
+
+class CompositeApi(BaseSalesforceApiTask):
+    task_docs = """
+This task is a wrapper for Composite REST API calls. Given a list of JSON files
+(one request body per file), POST each and process the returned composite
+result. Files are processed in the order given by the ``data_files`` option.
+
+In addition, this task will process the request body and replace namespace
+(``%%%NAMESPACE%%%``) and user ID (``%%%USERID%%%``) tokens. To avoid username
+collisions, use the ``randomize_username`` option to replace the top-level
+domains in any ``Username`` field with a random string.
+
+When the top-level ``allOrNone`` property for the request is set to true a
+SalesforceException is raised if an error is returned for any subrequest,
+otherwise partial successes will not raise an exception.
+
+Example Task Definition
+-----------------------
+
+.. code-block::  yaml
+
+  tasks:
+      example_composite_request:
+          class_path: cumulusci.tasks.salesforce.composite.CompositeApi
+          options:
+             data_files:
+                 - "datasets/composite/users.json"
+                 - "datasets/composite/setup_objects.json"
+    """
+
+    task_options = {
+        "data_files": {
+            "description": "A list of paths, where each path is a JSON file containing a composite request body.",
+            "required": True,
+        },
+        "managed": {
+            "description": "If True, replaces namespace tokens with the namespace prefix.",
+            "default": False,
+        },
+        "namespaced": {
+            "description": "If True, replaces namespace tokens with the namespace prefix.",
+            "default": False,
+        },
+        "randomize_username": {
+            "description": "If True, randomize the TLD for any 'Username' fields.",
+            "default": False,
+        },
+    }
+
+    def _init_options(self, kwargs):
+        super()._init_options(kwargs)
+        self.data_files = process_list_arg(self.options["data_files"])
```

Comment:
```suggestion
        self.data_files = process_list_arg(self.options.get("data_files") or [])
```

(_init_options runs before _validate_options which is where required options are checked, so it's not guaranteed to be there yet)